### ISSUE TITLE
[BUG] this.IsValid() updated after ReactiveCommand is invoked

### ISSUE THREAD

**Issue by antoine-l** [reactiveui/ReactiveUI.Validation#92]
**Describe the bug**
When using this.IsValid() as the argument for CanExecute in a ReactiveCOmmand.Create(), the IsValid is one user input "late".

**Steps To Reproduce**
This is a simple ViewModel.cs to reproduce the problem, for testing I bind it to a simple WPF TextBox:

```
    public class ViewModel : ReactiveValidationObject<ViewModel>
    {
        private readonly ReactiveCommand<string, Unit> Command;

        private string _myProperty;
        public string MyProperty
        {
            get => _myProperty;
            set => this.RaiseAndSetIfChanged(ref _myProperty, value);
        }

        public ViewModel()
        {
            Command = ReactiveCommand.Create<string>(x => SetMyProperty(x), this.IsValid());

            this
                .WhenAnyValue(x => x.MyProperty)
                .Do(x => Console.WriteLine("Property value = " + x))
                .InvokeCommand(Command);

            this.ValidationRule(
                viewModel => viewModel.MyProperty,
                x => !string.IsNullOrEmpty(x) && x.Length > 3,
                "Enter a value");

            this.IsValid()
                .Subscribe(x => Console.WriteLine("IsValid = " + x));
        }

        private Unit SetMyProperty(string value)
        {
            Console.WriteLine("Entered method");
            return Unit.Default;
        }
    }
```

This is the Console output: (notice when the command is executed)
Property value = 1
Property value = 12
Property value = 123
Property value = 1234
IsValid = True
Property value = 12345
Entered method
Property value = 1234
Entered method
Property value = 123
Entered method
IsValid = False
Property value = 12
Property value = 1

**Expected behavior**
This is the console output I would expect:
Property value = 1
Property value = 12
Property value = 123
Property value = 1234
IsValid = True
Entered method
Property value = 12345
Entered method
Property value = 1234
Entered method
Property value = 123
IsValid = False
Property value = 12
Property value = 1

**Environment**
- OS:      Windows 10, WPF App .NET Framework
- Version:   Validation=1.5.5, ReactiveUI = 11.5.35

Thank you so much for your help!

**Comment by worldbeater** [comment]
If you place a call to `this.WhenAnyValue`  below a call to `this.ValidationRule`, you'll get the following behavior:

```
IsValid = False
Property value = 1
Property value = 12
Property value = 123
IsValid = True
Property value = 1234
Entered method
Property value = 12345
Entered method
Property value = 1234
Entered method
IsValid = False
Property value = 123
Property value = 12
Property value = 1
```

The modified code looks as such:

```cs
public class ViewModel : ReactiveValidationObject<ViewModel>
{
    private readonly ReactiveCommand<string, Unit> Command;

    private string _myProperty;
    public string MyProperty
    {
        get => _myProperty;
        set => this.RaiseAndSetIfChanged(ref _myProperty, value);
    }

    public ViewModel()
    {
        Command = ReactiveCommand.Create<string>(x => SetMyProperty(x), this.IsValid());

        this.ValidationRule(
            viewModel => viewModel.MyProperty,
            x => !string.IsNullOrEmpty(x) && x.Length > 3,
            "Enter a value");
        
        // The call to WhenAny is now placed below a call to ValidationRule.
        this.WhenAnyValue(x => x.MyProperty)
            .Do(x => Console.WriteLine("Property value = " + x))
            .InvokeCommand(Command);

        this.IsValid()
            .Subscribe(x => Console.WriteLine("IsValid = " + x));
    }

    private Unit SetMyProperty(string value)
    {
        Console.WriteLine("Entered method");
        return Unit.Default;
    }
}
```

Both `WhenAny` and `ValidationRule` are using `CurrentThreadScheduler` since https://github.com/reactiveui/ReactiveUI.Validation/pull/97, so the call order matters now.

**Comment by antoine-l** [comment]
If I understand, the solution you are proposing will only work once the new NuGet package is deployed?

**Comment by glennawatson** [comment]
Yep he indicated since #97 and the release won't be immediate and will be a few days at least.

**Comment by worldbeater** [comment]
`ReactiveUI.Validation` with the fix is now shipped to NuGet as `^1.6.4`.

**Comment by antoine-l** [comment]
It works with 1.6.4. Thank you very much!
